types/auth0/index.d.ts

```diff
@@ -77,6 +76,10 @@ export interface GetUsersData {
   search_engine?: string;
 }
 
+export interface GetUsersDataPaged extends GetUsersData {
+  include_totals: boolean;
```

Comment:
Should the `per_page` and `page` properties also be moved here?

Reply:
The only thing I know (from [here](https://github.com/auth0/node-auth0/issues/276#issuecomment-398515690)) is that adding `include_totals` will make Auth0 return a paged result rather than an array. What other flags will reliably invoke this behaviour, I cannot say.

Reply:
But they are still parameters to a paged request, and it seems like they have no place in the main interface if you're not requesting paged data

Reply:
They would likely remain as optional properties. Do they add any value for non-paged requests? If not, they should likely only be passed with paged requests.

Reply:
If you move `per_page` and `page` to `GetUsersPaged` as optional properties, you will trigger the inference for the paged search path even when you fail to specify the `include_totals` flag. TSC will tell you will receive a pageset but at runtime you will actually get an array. Worse, existing projects that correctly assume arrays in this case will now have compile errors.

I think this is not so much a question of how we would like to design interfaces so much as how to make TSC correctly handle a very convoluted and (frankly excessively) polymorphic method signature.